torch/compiler/__init__.py

```diff
@@ -0,0 +1,221 @@
+import torch
+import torch._dynamo
+import torch._inductor
+from typing import Callable, Union, List, Set, Tuple, Any, Dict
+
+__all__ = [
+    "compile",
+    "is_enabled",
+    "reset",
+    "allow_in_graph",
+    "list_backends",
+    "explain",
+    "disable",
+]
+
+def compile(*args, **kwargs):
+    """
+    See :func:`torch.compile` for details on the arguments for this function.
+    """
+
+    return torch.compile(*args, **kwargs)
+
+def reset() -> None:
+    """
+    This function clears all compilation caches and restores the system to its initial state.
+    It is recommended to call this function, especially after using operations like `torch.compile(...)`
+    to ensure a clean state before subsequent compilation.
+
+    Usage:
+    1. Call `reset()` to clear all compilation caches and restore the initial state.
+    2. Perform any desired operations, such as `torch.compile(...)`.
+    3. If you need to start fresh or perform another `torch.compile(...)`, call `reset()` to ensure a clean state.
+
+    """
+
+    torch._dynamo.reset()
+
+def allow_in_graph(fn):
+    """
+    Customize which functions compilation will include in the generated graph.
+    Similar to `torch.fx.wrap()` it can be used as an escape hatch from graph breaks
```

Comment:
This is not entirely correct in what this does. 

It is not an escape hatch from graph breaks - it bypasses all introspection of the symbolic python code in favor of directly writing it to the graph. 

Furthermore, make_fx doe not honor it, and so you will still end up make_fxing the contents here, except without the dynamo safety guarantees.

Reply:
n00b q: So why would someone want to use this?